include/swift/Sema/ConstraintSystem.h

```diff
@@ -2301,6 +2301,8 @@ class ConstraintSystem {
   llvm::SmallMapVector<PackElementExpr *, PackExpansionExpr *, 2>
       PackEnvironments;
 
+  llvm::SmallVector<GenericEnvironment * , 4> PackElementGenericEnvironments;
```

Comment:
You'd have to add the same thing to Solution and use SolverScope to control addition, removal of these environments just like we do with i.e. `OpenedExistentialTypes`.

Reply:
Thank you! I saw that, but I thought that's only needed for maps😅